### PR TITLE
endsWith, startsWith, latinize, stripTags, wrap, stringular pipes

### PR DESCRIPTION
**Pull request by Iraecio**
More someones pipes of string.
- EndsWithPipe -> Check end of string and return boolean
- StartsWithPipe -> Check start of string and return boolean
- LatinizePipe -> Remove accents of string
- StripTagsPipe -> Remove tags of string make plain text
- WrapPipe -> Wraper string with someone term
- StringularPipe -> replace multiples arguments in string

All with docs and tests

### CODE REVIEW

**Comment by fknop** [comment]
The Stringular pipe repeats itself with the Template Pipe.

File: README.md [hunk 1]
```diff
@@ -98,6 +98,13 @@ You can find the documentations in the [`docs`](./docs) folder.
 * [`encodeURIComponent`](./docs/string.md#encodeuricomponent)
 * [`repeat`](./docs/string.md#repeat)
 * [`truncate`](./docs/string.md#truncate)
+* [`slugify`](./docs/string.md#slugify)
+* [`endsWith`](./docs/string.md#endsWith)
```
Comment (fknop) [inline README.md]: The markdown link after the `#` character does not take uppercase character. Write it all in lowercase.

File: README.md [hunk 2]
```diff
@@ -98,6 +98,13 @@ You can find the documentations in the [`docs`](./docs) folder.
...
+* [`stripTags`](./docs/string.md#stripTags)
```
Comment (fknop) [inline README.md]: The markdown link after the # character does not take uppercase character. Write it all in lowercase.

File: README.md [hunk 3]
```diff
@@ -98,6 +98,13 @@ You can find the documentations in the [`docs`](./docs) folder.
...
+* [`latinize`](./docs/string.md#latinize)
+* [`startWith`](./docs/string.md#startWith)
```
Comment (fknop) [inline README.md]: The markdown link after the # character does not take uppercase character. Write it all in lowercase.

File: docs/string.md [hunk 4]
```diff
@@ -16,6 +16,13 @@
 * [`encodeURIComponent`](#encodeuricomponent) 
 * [`repeat`](#repeat) 
 * [`truncate`](#truncate) 
+* [`slugfy`](#slugfy) 
+* [`endsWith`](#endsWith)
```
Comment (fknop) [inline docs/string.md]: The markdown link after the # character does not take uppercase character. Write it all in lowercase.

File: docs/string.md [hunk 5]
```diff
@@ -16,6 +16,13 @@
 * [`encodeURIComponent`](#encodeuricomponent) 
 * [`repeat`](#repeat) 
 * [`truncate`](#truncate) 
+* [`slugfy`](#slugfy) 
+* [`endsWith`](#endsWith) 
+* [`stripTags`](#stripTags)
```
Comment (fknop) [inline docs/string.md]: The markdown link after the # character does not take uppercase character. Write it all in lowercase.

File: docs/string.md [hunk 6]
```diff
@@ -16,6 +16,13 @@
 * [`encodeURIComponent`](#encodeuricomponent) 
 * [`repeat`](#repeat) 
 * [`truncate`](#truncate) 
+* [`slugfy`](#slugfy) 
+* [`endsWith`](#endsWith) 
+* [`stripTags`](#stripTags) 
+* [`latinize`](#latinize)
+* [`startWith`](#startWith)
```
Comment (fknop) [inline docs/string.md]: The markdown link after the # character does not take uppercase character. Write it all in lowercase.

File: src/string/starts-with.pipe.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+import {Pipe, PipeTransform} from '@angular/core';
+import {isString, isUndefined} from '../utils/utils';
+
+@Pipe({
+    name: 'startsWith'
+})
+export class StartsWithPipe implements PipeTransform {
+
+    transform(input: string, start: string, csensitive: boolean): any {
```
Comment (fknop) [inline src/string/starts-with.pipe.ts]: Use optional argument

File: src/string/latinize.pipe.ts [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+
+@Pipe({name: 'latinize'})
+export class LatinizePipe implements PipeTransform {
+
+    transform(input: string): any {
+
+        if (!isString(input) || isUndefined(input))
+            return input;
+
+        var replacementList = [
```
Comment (fknop) [inline src/string/latinize.pipe.ts]: Do one line per base/chars object.

File: src/string/latinize.pipe.ts [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+        for (var i = 0; i < replacementList.length; i += 1) {
+            var chars = replacementList[i].chars;
+            for (var j = 0; j < chars.length; j += 1) {
+                diacriticsMap[chars[j]] = replacementList[i].base;
+            }
+        }
+
+
+        return input.replace(/[^\u0000-\u007e]/g, (c) => diacriticsMap[c] || c);
+
```
Comment (fknop) [inline src/string/latinize.pipe.ts]: Remove empty line after return

File: src/string/stringular.pipe.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+// idea from https://github.com/a8m/angular-filter
+
+import {Pipe, PipeTransform} from '@angular/core';
+import {isString, isUndefined} from '../utils/utils';
+
+@Pipe({
+    name: 'stringular'
+})
+export class StringularPipe implements PipeTransform {
```
Comment (fknop) [inline src/string/stringular.pipe.ts]: This pipe is redundant with TemplatePipe.

File: src/string/wrap.pipe.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+import {Pipe, PipeTransform} from '@angular/core';
+import {isString, isUndefined} from '../utils/utils';
+
+@Pipe({
+    name: 'wrap'
+})
+export class WrapPipe implements PipeTransform {
+
+    transform(input: string, wrap: string, ends?: string): string {
+        return (!isString(input) || !isUndefined(wrap))
```
Comment (fknop) [inline src/string/wrap.pipe.ts]: `isString(input) && !isUndefined(wrap)`

File: docs/string.md [hunk 12]
```diff
@@ -291,3 +298,136 @@ import { TruncatePipe } from 'angular-pipes/src/string/truncate.pipe';
...
+####endsWith
+
+return whether string ends with the ends parameter.
+
+Arguments: ( string, ends, case-sensitive[optional] )
+
+##### File
+
+```
+import { EndsWithPipe } from 'angular-pipes/src/string/endsWith.pipe';
```
Comment (fknop) [inline docs/string.md]: The file is `ends-with.pipe` and not `endsWith.pipe`

File: src/string/ends-with.pipe.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+@Pipe({
+    name: 'endsWith'
+})
+export class EndsWithPipe implements PipeTransform {
+
+    transform(input: string, ends: string, csensitive?: boolean): any {
+
+        let sensitive: boolean = csensitive || false,
+            position: any;
```
Comment (fknop) [inline src/string/ends-with.pipe.ts]: Declare position where it is first used. + 1 let/const per variable

File: src/string/ends-with.pipe.ts [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+import {isString, isUndefined} from '../utils/utils';
+
+@Pipe({
+    name: 'endsWith'
+})
+export class EndsWithPipe implements PipeTransform {
+
+    transform(input: string, ends: string, csensitive?: boolean): any {
+
+        let sensitive: boolean = csensitive || false,
```
Comment (fknop) [inline src/string/ends-with.pipe.ts]: Default parameter instead of optional will avoid doing this

File: src/string/ends-with.pipe.ts [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+
+import {Pipe, PipeTransform} from '@angular/core';
+import {isString, isUndefined} from '../utils/utils';
+
+@Pipe({
+    name: 'endsWith'
+})
+export class EndsWithPipe implements PipeTransform {
+
+    transform(input: string, ends: string, csensitive?: boolean): any {
```
Comment (fknop) [inline src/string/ends-with.pipe.ts]: `csensitive: boolean = false` - See comment below

File: docs/string.md [hunk 16]
```diff
@@ -291,3 +298,136 @@ import { TruncatePipe } from 'angular-pipes/src/string/truncate.pipe';
...
+
+```html
+{{ 'image.JPG' | endsWith: '.jpg' }} <!-- result: True -->
+{{ 'image.JPG' | endsWith: '.jpg': true }} <!-- result: False -->
+```
+
+####stripTags
+
+strip out html tags from string
+**Important: this Pipe jobs it's not to replace ng-bind-html directive, it's only for tiny plain text
```
Comment (fknop) [inline docs/string.md]: Missing ** for bold end

File: src/string/ends-with.pipe.ts [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+            position: any;
+
+        if (!isString(input) || isUndefined(ends)) {
+            return input;
+        }
+
+        input = (sensitive) ? input : input.toLowerCase();
+        position = input.length - ends.length;
+
+        return input.indexOf((sensitive) ? ends : ends.toLowerCase(), position) !== -1;
```
Comment (fknop) [inline src/string/ends-with.pipe.ts]: Maybe you should add startsWith / endsWith function in the utils file and import them

File: src/string/latinize.pipe.ts [hunk 18]
```diff
@@ -0,0 +1,325 @@
...
+import {Pipe, PipeTransform} from '@angular/core';
+import {isString, isUndefined} from '../utils/utils';
+
+@Pipe({name: 'latinize'})
+export class LatinizePipe implements PipeTransform {
+
+    transform(input: string): any {
+
+        if (!isString(input) || isUndefined(input))
```
Comment (fknop) [inline src/string/latinize.pipe.ts]: Add `{ }`

File: src/string/starts-with.pipe.ts [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+
+        let sensitive = csensitive || false;
```
Comment (fknop) [inline src/string/starts-with.pipe.ts]: Remove this with default argument

File: src/string/starts-with.pipe.ts [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+
+        if (!isString(input) || isUndefined(start))
```
Comment (fknop) [inline src/string/starts-with.pipe.ts]: Add `{ }`

File: src/string/ends-with.pipe.ts (hunk 17 above)
Comment (Iraecio) [inline src/string/ends-with.pipe.ts]: one function with two argument start and end

File: docs/string.md [hunk 21]
```diff
@@ -16,6 +16,11 @@
 * [`encodeURIComponent`](#encodeuricomponent) 
 * [`repeat`](#repeat) 
 * [`truncate`](#truncate) 
+* [`slugfy`](#slugfy)
```
Comment (fknop) [inline docs/string.md]: Missing `i`

File: docs/string.md [hunk 22]
```diff
@@ -291,3 +296,109 @@ import { TruncatePipe } from 'angular-pipes/src/string/truncate.pipe';
...
+####slugify
+
+Slugify a string. 
+
+Arguments: (string)
+
+##### File
+
+```
+import { SlugifyPipe } from 'angular-pipes/src/string/slugfy.pipe';
```
Comment (fknop) [inline docs/string.md]: Missing `i`

File: src/string/latinize.pipe.ts [hunk 23]
```diff
@@ -0,0 +1,126 @@
...
+        for (let i = 0; i < replacementList.length; i += 1) {
+            let chars = replacementList[i].chars;
+            for (let j = 0; j < chars.length; j += 1) {
+                diacriticsMap[chars[j]] = replacementList[i].base;
+            }
+        }
+
+        return input.replace(/[^\u0000-\u007e]/g, (c) => diacriticsMap[c] || c);
+    }
+
```
Comment (fknop) [inline src/string/latinize.pipe.ts]: Remove empty line

**Comment by fknop** [comment]
You need to pull the master branch to resolve conflicts.

**Pull request closed without merging**